impala/hiveserver2.py

```diff
@@ -259,11 +259,13 @@ def close(self):
         if exc_info:
             six.reraise(*exc_info)
 
-    def cancel_operation(self):
+    def cancel_operation(self, reset_state=True):
         if self._last_operation_active:
             log.info('Canceling active operation')
-            self._last_operation.cancel()
-            self._reset_state()
+            resp = self._last_operation.cancel()
+            if reset_state:
+                self._reset_state()
+            return resp
```

Comment:
Do we need to expose this? Seems like we don't want clients to start to depend on this. Based on your other change this might be leftover debugging code...

Reply:
I removed the return code but left the return statements inside Operation() below for consistency. Let me know if you'd prefer those removed, too.